src/parsers/perf/perfparser.cpp

```diff
@@ -999,6 +999,27 @@ class PerfParserPrivate : public QObject
 
     void addSample(const Sample& sample)
     {
+        auto& lastTime = m_lastSampleTimePerCore[sample.cpu];
```

Comment:
please put that into a separate function, which you then call from here. that allows you to simplify the code as such:

```
auto& lastTime = m_lastSampleTimePerCore[sample.cpu];
auto updateLastTime = qScopeGuard([&]() { lastTime = sample.time; });
if (!lastTime) {
    return;
}

if (frequencyResult.cores.size() <= sample.cpue) {
    frequencyResult.cores.resize(sample.cpu + 1);
}

auto &core = frequencyResult.cores[sample.cpu];
// then use core below instead of looking it up all the time
```